packages/core/src/routes/interaction/verifications/mfa-verification.ts

```diff
@@ -56,6 +61,39 @@ export const verifyMfa = async (
 
   const { mfaVerifications } = await tenant.queries.users.findUserById(accountId);
 
+  const availableFactors = mfaVerifications
+    .filter((verification) => {
+      if (verification.type !== MfaFactor.BackupCode) {
+        return true;
+      }
+
+      // Skip backup code if it is used
+      return verification.codes.some((code) => !code.usedAt);
+    })
+    .reduce<MfaVerification[]>((factors, verification) => {
+      // Ingnore duplicated verification
+      if (factors.some(({ type }) => type === verification.type)) {
+        return factors;
+      }
+
+      return [...factors, verification];
+    }, [])
```

Comment:
could merge using one single reduce?

Reply:
I believe by spilting "filter" and "reduce", it would be easier to read the code.